### PR TITLE
Backports 0.19 pr10

### PR DESCRIPTION
**Pull request by Munkybooty**

### CODE REVIEW

File: doc/bips.md [hunk 1]
```diff
@@ -18,15 +18,20 @@ BIPs that are implemented by Bitcoin Core (up-to-date up to **v0.19.0**):
 * [`BIP 61`](https://github.com/bitcoin/bips/blob/master/bip-0061.mediawiki): The 'reject' protocol message (and the protocol version bump to 70002) was added in **v0.9.0** ([PR #3185](https://github.com/bitcoin/bitcoin/pull/3185)). Starting *v0.16.0*, whether to send reject messages can be configured with the `-enablebip61` option.
 * [`BIP 65`](https://github.com/bitcoin/bips/blob/master/bip-0065.mediawiki): The CHECKLOCKTIMEVERIFY softfork was merged in **v0.12.0** ([PR #6351](https://github.com/bitcoin/bitcoin/pull/6351)), and backported to **v0.11.2** and **v0.10.4**. Mempool-only CLTV was added in [PR #6124](https://github.com/bitcoin/bitcoin/pull/6124).
 * [`BIP 66`](https://github.com/bitcoin/bips/blob/master/bip-0066.mediawiki): The strict DER rules and associated version 3 blocks have been implemented since **v0.10.0** ([PR #5713](https://github.com/bitcoin/bitcoin/pull/5713)).
-* [`BIP 68`](https://github.com/bitcoin/bips/blob/master/bip-0068.mediawiki): Sequence locks have been implemented as of **v0.12.1**  ([PR #7184](https://github.com/bitcoin/bitcoin/pull/7184)), and have been activated since *block 419328*.
```
Comment (PastaPastaPasta) [inline doc/bips.md:21]: 17197: DNM don't do PRs when bitcoin does backports to an earlier version (unless strictly necessary)

File: src/Makefile.am [hunk 2]
```diff
@@ -869,13 +869,13 @@ clean-local:
 check-symbols: $(bin_PROGRAMS)
 if GLIBC_BACK_COMPAT
 	@echo "Checking glibc back compat..."
-	$(AM_V_at) READELF=$(READELF) CPPFILT=$(CPPFILT) $(PYTHON) $(top_srcdir)/contrib/devtools/symbol-check.py < $(bin_PROGRAMS)
```
Comment (PastaPastaPasta) [inline src/Makefile.am:872]: 17858: again

File: src/test/cuckoocache_tests.cpp [hunk 3]
```diff
@@ -13,6 +13,7 @@
 #include <shared_mutex>
 #include <thread>
 #include <vector>
+#include <deque>
```
Comment (PastaPastaPasta) [inline src/test/cuckoocache_tests.cpp:16]: 17988: again, also this breaks linters

File: src/qt/walletview.h [hunk 4]
```diff
@@ -73,7 +73,7 @@ class WalletView : public QStackedWidget
 
     TransactionView *transactionView;
 
-    QProgressDialog *progressDialog;
```
Comment (PastaPastaPasta) [inline src/qt/walletview.h:76]: 18084: again re bitcoin backports

File: src/psbt.cpp [hunk 5]
```diff
@@ -58,8 +58,11 @@ bool PartiallySignedTransaction::AddOutput(const CTxOut& txout, const PSBTOutput
 bool PartiallySignedTransaction::GetInputUTXO(CTxOut& utxo, int input_index) const
 {
     PSBTInput input = inputs[input_index];
-    int prevout_index = tx->vin[input_index].prevout.n;
```
Comment (PastaPastaPasta) [inline src/psbt.cpp:61]: 18079: again is a btc backport

**Review by PastaPastaPasta — CHANGES_REQUESTED** [review]

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: doc/bips.md [hunk 6]
```diff
@@ -19,13 +19,14 @@ BIPs that are implemented by Bitcoin Core (up-to-date up to **v0.18.0**):
...
-* [`BIP 70`](https://github.com/bitcoin/bips/blob/master/bip-0070.mediawiki) [`71`](https://github.com/bitcoin/bips/blob/master/bip-0071.mediawiki) [`72`](https://github.com/bitcoin/bips/blob/master/bip-0072.mediawiki): Payment Protocol support has been available in Bitcoin Core GUI since **v0.9.0** ([PR #5216](https://github.com/bitcoin/bitcoin/pull/5216)). Support can be optionally disabled at build time since **v0.18.0** ([PR 4350](https://github.com/dashpay/dash/pull/4350)).
+* [`BIP 70`](https://github.com/bitcoin/bips/blob/master/bip-0070.mediawiki) [`71`](https://github.com/bitcoin/bips/blob/master/bip-0071.mediawiki) [`72`](https://github.com/bitcoin/bips/blob/master/bip-0072.mediawiki): Payment Protocol support has been available in Bitcoin Core GUI since **v0.9.0** ([PR #5216](https://github.com/bitcoin/bitcoin/pull/5216)). Support can be optionally disabled at build time since **v0.18.0** ([PR 4350](https://github.com/dashpay/dash/pull/4350)), and is disabled by default at build time since **v0.19.0** ([PR #15584](https://github.com/bitcoin/bitcoin/pull/15584)).
```
[the remark this answers is not in the document]
Comment (Munkybooty) [reply doc/bips.md:22]: would i just be better off just saying that it is enabled by default?
 I'm not sure which version number to write for it or pr if i were to do that.

**Comment by Munkybooty** [comment]
whoops, i added back too much

File: build-aux/m4/ax_boost_base.m4 [hunk 7]
```diff
@@ -127,8 +127,14 @@ AC_DEFUN([_AX_BOOST_BASE_RUNDETECT],[
...
 
+    dnl some arches may advertise a cpu type that doesn't line up with their
+    dnl prefix's cpu type. For example, uname may report armv7l while libs are
+    dnl installed to /usr/lib/arm-linux-gnueabihf. Try getting the compiler's
+    dnl value for an extra chance of finding the correct path.
+    libsubdirs="lib/`$CXX -dumpmachine 2>/dev/null` $libsubdirs"
+
     dnl first we check the system location for boost libraries
-    dnl this location ist chosen if boost libraries are installed with the --layout=system option
+    dnl this location is chosen if boost libraries are installed with the --layout=system option
```
Comment (UdjinM6) [inline build-aux/m4/ax_boost_base.m4:137]: 432098c99381e564a3796663510534ed48e76e50: this patch was never merged into master and the issue was fixed via 18558 (which we backported already), see 18501

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: doc/bips.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (UdjinM6) [reply doc/bips.md:22]: 15584 wasn't backported yet. I'd just drop 16997 for now.

**Review by UdjinM6 — APPROVED** [review]
utACk

**Review by PastaPastaPasta — APPROVED** [review]
utACK for merging via merge commit

**Pull request merged**